benchmark/utils/runner.hpp

```diff
@@ -43,10 +46,55 @@ struct Benchmark {
 };
 
 
+namespace detail {
+
+
+inline void expand(const json& elem,
+                   const std::vector<std::pair<std::string, json>>& lists,
+                   int depth, json& result)
+{
+    if (depth == lists.size()) {
+        result.push_back(elem);
+    } else {
+        auto [name, next_list] = lists[depth];
+        for (const auto& next_elem : next_list) {
+            json combined_elem = elem;
+            combined_elem[name] = next_elem;
+            expand(combined_elem, lists, depth + 1, result);
+        }
+    }
+}
+
+
+}  // namespace detail
+
+
+/**
+ * Creates a new JSON array object by taking the cartesian product of all
+ * values of the input dict. Any value that isn't a list will be converted into
+ * a list of a single item.
```

Comment:
```suggestion
 * values of the input dict. Any value that isn't a list will be considered as
 * a list of a single item during expanding.
```
I was thinking it is produce `"A": ["B"]` in the resulting file from `"A": "B"`, but it is considered as `"A": ["B"]` during processing and lead "A": "B" in all, right?

Reply:
Consider this input:
```
{
  "a": "b",
  "c": ["d"]
}
```
it will produce
```
[
  {
    "a": "b",
    "c": "d"
  }
]
```
Does that answer your question?